### PR TITLE
refactor(button): refactor code to use @each

### PR DESCRIPTION
**Pull request by rands0n**
Use `@each` instead create more classes

### CODE REVIEW

**Comment by fernandofleury** [comment]
image

**Comment by leandrogrillo** [comment]
i was thinking about using `&-` selector. 

Based on postcss "ideology", we shouln't use this selector because it's planned to be in next css version, because this postcss plugin take the same ideia form sass.

So, isn't better using `.selector-$color` instead of "sass" selector?

@fernandofleury @rands0n

File: src/css/atoms/button.css [hunk 1]
```diff
@@ -64,43 +66,39 @@
...
 
-  &-text {
-    background-color: transparent;
-    color: $color-primary;
-    padding: 0;
-    transition: none;
+.button-neutral {
+  background-color: $color-neutral;
+  color: $color-text;
+}
```
Comment (leandrogrillo) [inline src/css/atoms/button.css]: is it right, two `.button-neutral` in this file?

Comment (rands0n) [inline src/css/atoms/button.css]: removed :)

Comment (leandrogrillo) [inline src/css/atoms/button.css]: ty

**Pull request merged**